web-core/src/main/java/com/tutrit/gateway/CustomerGateway.java

```diff
@@ -0,0 +1,11 @@
+package com.tutrit.gateway;
+
+import com.tutrit.bean.Customer;
+
+import java.util.Optional;
+
+public interface CustomerGateway {
+    Optional<Customer> saveCustomer (Customer customer);
```

Comment:
that will definitely return new Customer, so no Optional here

Reply:
I agree, I've redone it